runtime/doc/options.txt

```diff
@@ -6254,6 +6254,14 @@ A jump table for the options with a short description can be found at |Q_op|.
 	   uselast	If included, jump to the previously used window when
 			jumping to errors with |quickfix| commands.
 
+						*'syncout'* *'nosyncout'*
+'syncout'		boolean	(default off)
```

Comment:
If this is permanently TUI-only, shouldn't we prefix it with `term`, like `termguicolors`? That helps discoverability.

```suggestion
'termsyncout'		boolean	(default off)
```
If there's a chance this may be supported by all UIs in the future , the usual convention is to prefix it with `gui`

Reply:
Actually, synchronized output makes sense for the terminal client as well as GUI clients. Why would the GUI not want to avoid tearing, if possible? How is this capability informed towards a potential GUI frontend?

Reply:
> Why would the GUI not want to avoid tearing, if possible? 

Both Nvim UI protocol and Nvim UIs already try to batch screen updates.

So this option is likely irrelevant for GUIs. This PR merely sets a terminal feature which is not present in all terminals. Such a concept is irrelevant for GUIs.

Reply:
> Why not default to "on"? (Does it even need to be optional?)
> 

My preference would also be for it to default to on and not be optional, but I was taking the conservative approach here. I’m fine with changing it if there’s no objection.

Reply:
A developer might want to have the possibility to have it disabled, to interactively see the screen redrawn as the developer single-steps through each call in the code.

From the user's perspective, there is **zero** need to have this configurable, tbh - the developer (terminal as well as nvim or even nvim plugin dev) might very well want to control it.

Reply:
With TUI stuff historically we try to enable as much by default as possible. Easy to revert later if we run into unsolvable problems.

Reply:
+1 to make it the default. That would match existing GUI behaviors so guis could interpret explicitly setting it to false if it wants to (or just ignore it )

Reply:
Sounds good 👍. If GUIs are meant to interpret this should we stick with 'syncout' over 'termsyncout' and make the help text more UI generic (rather than TUI specific)?